frontend/src/APIClients/BaseAPIClient.ts

```diff
@@ -46,4 +46,15 @@ baseAPIClient.interceptors.request.use(async (config: AxiosRequestConfig) => {
   return newConfig;
 });
 
+baseAPIClient.interceptors.response.use(
+  (res) => res,
+  (error) => {
+    if (error.response.status === 401) {
```

Comment:
We want to first try refreshing the token upon a `401` (already implemented in the interceptor above) and only logout if that fails. Are you sure this doesn't skip the refresh step?